e2e/specs/st_slider.spec.js

```diff
@@ -18,20 +18,23 @@
 describe("st.slider", () => {
   beforeEach(() => {
     cy.visit("http://localhost:3000/");
-    // Open sidebar expander
-    cy.get(".streamlit-expanderHeader").click();
   });
 
   it("looks right", () => {
     // Make the ribbon decoration line disappear
     cy.get("[data-testid='stDecoration']").invoke("css", "display", "none");
 
     cy.get(".stSlider")
-      .eq(2)
+      .eq(1)
       .matchThemedSnapshots("slider");
   });
 
   it("shows labels", () => {
+    // Open sidebar expander
+    cy.wait(1000);
+    cy.get(".streamlit-expanderHeader").click();
+    cy.wait(1000);
+
```

Comment:
I think you can use `cy.contains` to write this in a way that doesn't depend on waiting a hard-coded amount of time:

```javascript
// Not sure if the first cy.wait(1000) is droppable here, but I think it should be?
cy.get(".streamlit-expanderHeader").click();
cy.get(".stSlider label").contains("Label B");

// the rest of the test...
```

should wait until "Label B" appears in the text of the selected elements before proceeding to run the next command, and if the given text doesn't appear by the timeout (default 4s), an error is thrown

Reply:
Agreed this is a much better approach - thank you! 
Also, without the first `cy.wait`, the `maintains its state on rerun` test was occasionally failing, but I have since tested it a bunch of times successfully, so looks like it is no longer necessary.